### PR TITLE
Enable Obstacle Layer on Global and Local costmaps

### LINKED ISSUE #613: Enable Obstacle Layer by Default
## Feature Request

The obstacle layer in costmap is off by default. It should be on. 

This may require #552

### PR DESCRIPTION
**Pull request by mhpanah**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | (fixes #613) |
| Primary OS tested on | (Ubuntu 18.04) |
| Robotic platform tested on | () |

---

## Description of contribution in a few bullet points

- Changed AMCL to not broadcast transforms before initial pose is received either via initialpose topic or later when pose from server is available.
    - This way robot knows it location with respect to map before putting obstacles on map.
- Enabled obstacle layers for both global and local costmaps

---

### CODE REVIEW

**Comment by mhpanah** [comment]
> If I understand this correctly, AMCL now won't try to publish a pose until it has received an initial pose, right? Or is it just it won't populate the obstacle layer? If it won't publish a pose, I don't think that's what we want, or the auto-localization won't work.

Yes, that's correct.  Actually, I think we should not make amcl to publish a pose that is not valid.  AMCL should get it's initial pose either from parameter server, file, user input, global localizer, auto-localization or etc. We should not run amcl if the initial conditions are not known.

File: nav2_amcl/src/amcl_node.cpp [hunk 1]
```diff
@@ -92,7 +92,8 @@ AmclNode::AmclNode()
 
   createMotionModel();
 
-  updatePoseFromServer();
+  // updatePoseFromServer();
```
[the remark this answers is not in the document]
Comment (mhpanah) [reply nav2_amcl/src/amcl_node.cpp]: This is a placeholder for getting initial pose from parameter server #290.

**Comment by mhpanah** [comment]
@mkhansen-intel pushed a fix for auto-localization.

**Comment by mhpanah** [comment]
Will merge this and open a new issue to make sure auto-localization and global localization service still works.

**Comment by mkhansen-intel** [comment]
I think you need to update the AMCL README and explain that it won't publish a pose until initial pose is set, as that was NOT the behavior in ROS and anyone who is used to using the ROS version won't know that is the case.

**Pull request merged**